internal/restic/snapshot.go

```diff
@@ -201,6 +201,17 @@ func (sn *Snapshot) HasPaths(paths []string) bool {
 	return true
 }
 
+// HasHostname returns true if the snapshot hostname is in the list of the given hostnames.
+func (sn *Snapshot) HasHostname(hostnames []string) bool {
```

Comment:
If `HasHostname()` accepts an empty slice and returns `true` in this case we can remove the `nil` checks below (needs to be mentioned in the description for the function though).

For slice emptyness, we should use `if len(hostnames) == 0`, this will match `hostnames == nil` as well as `hostnames == []string{}`.

Reply:
is changed, see above.